src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2ConnectionTests.cs

```diff
@@ -3618,8 +3617,7 @@ public async Task RST_STREAM_IncompleteRequest_AdditionalWindowUpdateFrame_Conne
         await SendRstStreamAsync(1);
         await SendWindowUpdateAsync(1, 1024);
 
-        await WaitForConnectionErrorAsync<Http2ConnectionErrorException>(ignoreNonGoAwayFrames: false, expectedLastStreamId: 1,
-            Http2ErrorCode.STREAM_CLOSED, CoreStrings.FormatHttp2ErrorStreamAborted(Http2FrameType.WINDOW_UPDATE, 1));
+        await WaitForStreamErrorAsync(expectedStreamId: 1, Http2ErrorCode.STREAM_CLOSED, CoreStrings.Http2StreamAborted);
 
         tcs.TrySetResult(); // Don't let the response start until after the abort
```

Comment:
Nice that we already had a test for this. Now that this isn't a connection error, I think we should assert we can close the connection gracefully.


```suggestion
        tcs.TrySetResult(); // Don't let the response start until after the abort

        await StopConnectionAsync(expectedLastStreamId: 1, ignoreNonGoAwayFrames: false);

        AssertConnectionNoError();
```